File: rb/features.rb
# OpenElevation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module OpenElevationFeatures
  def self.make_feature(name)
    case name
    when "base"
      OpenElevationBaseFeature.new
    when "test"
      OpenElevationTestFeature.new
    else
      OpenElevationBaseFeature.new
    end
  end
end

File: py/openelevation_sdk/features.py
# OpenElevation SDK feature factory

from openelevation_sdk.feature.base_feature import OpenElevationBaseFeature
from openelevation_sdk.feature.test_feature import OpenElevationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenElevationBaseFeature(),
        "test": lambda: OpenElevationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
